silkworm/rpc/commands/ots_api.cpp

```diff
@@ -877,18 +877,19 @@ Task<TransactionsWithReceipts> OtsRpcApi::collect_transactions_with_receipts(
         SILK_DEBUG
             << "txn_id: " << tnx_nums->txn_id
             << " block_num: " << tnx_nums->block_num
-            << ", txn_index: " << tnx_nums->txn_index
+            << ", txn_index: " << tnx_nums->txn_index.value_or(-1)
+            << ", initial txn: " << tnx_nums->initial_txn
             << ", final txn: " << tnx_nums->final_txn
             << ", ascending: " << std::boolalpha << ascending;
 
-        if (tnx_nums->final_txn) {
-            continue;
-        }
-
         if (tnx_nums->block_changed) {
             block_info.reset();
         }
 
+        if (tnx_nums->initial_txn || tnx_nums->final_txn) {
```

Comment:
`initial_txn` and `final_txn` can be removed since `txn_index` is now optional

Reply:
ok